src/plugins/sppPlugin/hooks/useSppPermissionCheckProposalCreation/useSppPermissionCheckProposalCreation.ts

```diff
@@ -32,12 +37,33 @@ export const useSppPermissionCheckProposalCreation = (
         })?.({ plugin, daoId, useConnectedUserInfo }),
     );
 
+    if (isSimulationLoading) {
+        return {
+            isLoading: true,
+            isRestricted: false,
+            hasPermission: false,
+            settings: [],
+        };
+    }
+
+    if (hasSimulationSucceeded) {
+        return {
+            isLoading: false,
+            isRestricted: false,
+            hasPermission: true,
+            settings: [],
+        };
+    }
+
     // Allow proposal creation if either:
-    // - All plugins are unrestricted.
-    // - User has permission on at least one restricted plugin.
-    const permissionGranted =
-        pluginProposalCreationGuardResults.every((result) => !result?.isRestricted) ||
-        pluginProposalCreationGuardResults.some((result) => result?.isRestricted && result.hasPermission);
+    // OLD LOGIC:
+    //   - All plugins are unrestricted.
+    //   - User has permission on at least one restricted plugin.
+    // NEW LOGIC:
+    //   - Simulation is successful.
+    //
```

Comment:
This comment feels like scratch pad notes, do we need it still? I don't really see the benefit of labeling OLD and NEW if the logic works for both.